pkg/skaffold/deploy/util/namespaces.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"sort"
+
+	kubectx "github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/context"
+	latestV1 "github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest/v1"
+)
+
+// GetAllPodNamespaces lists the namespaces that should be watched.
+// + The namespace passed on the command line
+// + Current kube context's namespace
+// + Namespaces referenced in Helm releases
+func GetAllPodNamespaces(configNamespace string, pipelines []latestV1.Pipeline) ([]string, error) {
+	nsMap := make(map[string]bool)
+
+	if configNamespace == "" {
+		// Get current kube context's namespace
+		config, err := kubectx.CurrentConfig()
+		if err != nil {
+			return nil, fmt.Errorf("getting k8s configuration: %w", err)
+		}
```

Comment:
How will this interact if different skaffold modules have different kubeContext defined?

Reply:
I'm not 100% sure but I think this runs once per pipeline, so it would handle different contexts per module fine. this was working before I made this change so I'm inclined to believe it's right. maybe @gsquared94 can confirm

Reply:
we have a config validation to ensure that all modules have the same `kubeContext` value. We don't yet support multi-cluster deployment.